### PR TITLE
update for titiler 0.4.0

### PR DESCRIPTION
**Pull request by vincentsarago**
This PR Does:

- up titiler requirement to `0.4.0` 

Note: new titiler version changed a lot of `query-parameter` style and name see: https://github.com/developmentseed/titiler/discussions/396

For now the Tests fails because there has been a change in pgstac behaviour. When we register a Search request with a bbox we were expecting that requesting a tile outside the bbox would return empty items ... but it's no more the case with `0.4.0` 🤷‍♂️  cc @bitner

### CODE REVIEW

File: tests/test_titiler_pgstac.py [hunk 1]
```diff
@@ -159,5 +159,6 @@ def test_tiles(rio, app):
     assert meta["width"] == 256
     assert meta["height"] == 256
 
+    # tile is outside mosaic bbox, it should return 404 (NoAssetFoundError)
     response = app.get(f"/tiles/{search_bbox}/{z}/{x}/{y}?assets=cog")
     assert response.status_code == 404
```
Comment (vincentsarago) [inline tests/test_titiler_pgstac.py:164]: This test is failing now. The tile requested is outside the mosaic bbox but still return assets now

Comment (bitner) [reply tests/test_titiler_pgstac.py:164]: pgstac v0.4.0 makes CQL2-json the default query language. All the tests are using the old query syntax which in pgstac get rewritten into CQL1. Adding "filter-lang":"cql-json" should allow these tests to still work until updated to CQL2-json.

Comment (vincentsarago) [reply tests/test_titiler_pgstac.py:168]: 🙏 thanks @bitner it fixed it 😄

**Comment by vincentsarago** [comment]
TODO:

- add tests with CQL-2

**Pull request merged**